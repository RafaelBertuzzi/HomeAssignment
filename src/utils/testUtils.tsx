import { ReactNode } from 'react';
import renderer from 'react-test-renderer';

import { LayoutProvider } from '@components/LayoutProvider';
import { render } from '@testing-library/react-native';

export const rendererWithProviders = (children: ReactNode) => {
  return renderer.create(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};

export const renderWithProviders = (children: ReactNode) => {
  return render(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};
