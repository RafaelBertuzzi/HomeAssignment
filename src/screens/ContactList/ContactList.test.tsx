import { Contact } from 'react-native-contacts';

import { cleanup } from '@testing-library/react-native';
import { rendererWithProviders } from '@utils/testUtils';

import { ContactList } from '.';

const mockedData = [
  {
    birthday: { day: 20, month: 1, year: 1978 },
    company: 'Creative Consulting',
    emailAddresses: [],
    familyName: 'Bell',
    givenName: 'Kate',
    hasThumbnail: false,
    imAddresses: [],
    jobTitle: 'Producer',
    middleName: '',
    phoneNumbers: [],
    postalAddresses: [],
    recordID: '177C371E-701D-42F8-A03B-C61CA31627F6',
    thumbnailPath: '',
    urlAddresses: [],
    backTitle: 'Back',
    department: 'Marketing',
    displayName: 'Kate Bell',
    isStarred: false,
    note: 'Kate Bell has been working on new designs for our holiday product line. She has some great ideas.',
    prefix: 'Ms.',
    suffix: 'Jr.',
  },
] as Contact[];

jest.mock('@features/ContactList/hooks/useContactList', () => ({
  useContactList: () => ({
    data: mockedData,
    isLoading: false,
    refetch: jest.fn(),
    isRefetching: false,
  }),
}));

describe('ContactList Screen', () => {
  afterEach(cleanup);

  it('should render ContactList Screen', () => {
    const json = rendererWithProviders(<ContactList />);
    expect(json).toMatchSnapshot();
  });
});
