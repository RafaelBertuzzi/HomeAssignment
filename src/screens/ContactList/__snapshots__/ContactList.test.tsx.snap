// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList Screen should render ContactList Screen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      dataSet={{}}
      style={
        {
          "position": "fixed",
        }
      }
    >
      <View
        dataSet={{}}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": 16,
            "marginRight": 16,
          }
        }
      >
        <Text
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 20,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 30,
              "textDecorationLine": undefined,
            }
          }
        >
          Contacts List
        </Text>
        <View
          dataSet={{}}
          style={{}}
        />
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          colorScheme="primary"
          dataSet={{}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 10,
            }
          }
        >
          <View
            dataSet={{}}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
            test={true}
          >
            <View
              dataSet={{}}
              style={{}}
            >
              <Image
                accessibilityLabel="search"
                alt="search"
                dataSet={{}}
                onError={[Function]}
                source={
                  {
                    "testUri": "../../../src/assets/icons/search.png",
                  }
                }
                style={
                  {
                    "height": 24,
                    "maxWidth": "100%",
                    "width": 24,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        aria-orientation="horizontal"
        dataSet={{}}
        style={
          {
            "backgroundColor": "#d4d4d4",
            "height": 1,
            "width": "100%",
          }
        }
        thickness="1"
      />
    </View>
    <View
      dataSet={{}}
      style={{}}
    />
    <View
      dataSet={{}}
      style={
        {
          "flex": 1,
          "marginLeft": 16,
          "marginRight": 16,
          "marginTop": 16,
        }
      }
    >
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        ListFooterComponent={[Function]}
        data={
          [
            {
              "backTitle": "Back",
              "birthday": {
                "day": 20,
                "month": 1,
                "year": 1978,
              },
              "company": "Creative Consulting",
              "department": "Marketing",
              "displayName": "Kate Bell",
              "emailAddresses": [],
              "familyName": "Bell",
              "givenName": "Kate",
              "hasThumbnail": false,
              "imAddresses": [],
              "isStarred": false,
              "jobTitle": "Producer",
              "middleName": "",
              "note": "Kate Bell has been working on new designs for our holiday product line. She has some great ideas.",
              "phoneNumbers": [],
              "postalAddresses": [],
              "prefix": "Ms.",
              "recordID": "177C371E-701D-42F8-A03B-C61CA31627F6",
              "suffix": "Jr.",
              "thumbnailPath": "",
              "urlAddresses": [],
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        refreshControl={
          <RefreshControlMock
            onRefresh={[MockFunction]}
            refreshing={false}
          />
        }
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={[]}
        viewabilityConfigCallbackPairs={[]}
      >
        <RCTRefreshControl />
        <View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={null}
          >
            <View
              dataSet={{}}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Kate Bell
              </Text>
            </View>
          </View>
          <View
            onLayout={[Function]}
          >
            <View
              dataSet={{}}
              style={
                {
                  "alignItems": "center",
                  "marginTop": 8,
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                1 contacts
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
